app/server/middleware/repos.js

```diff
@@ -8,54 +8,45 @@ module.exports = {
         console.log('listing repos [' + req.path + ']');
         console.log('query:' + JSON.stringify(req.query, null, 2));
 
-        repoService.getRepos().then((repos) => {
-            req.entity = repos.map((repo) => {
-                return {
-                    'id': repo.id,
-                    'name': repo.name,
-                    'description': repo.description,
-                    'public': !repo.private
-                };
-            });
-
+        repoRepository.getRepos().then(repos => {
+            req.entity = repos;
             next();
-
-        }).catch((err) => {
-            next(err);
-        });
+        }).catch(err => next(err));
     },
 
     listReposPermission (req, res, next) {
         console.log('looking up user permission for repos');
 
-        let user = req.query.permission_user,
+        let username = req.query.permission_user,
             repos = req.entity;
 
-        if (user) {
-            repos.forEach((repo) => {
-                repo.permission = 'read';
-            });
+        if (username) {
+            repoRepository.getPermissions(username, repos).then(repos => {
+                req.entity = repos;
+                next();
+            }).catch(err => next(err));
+        } else {
+            next();
         }
-        next();
     },
 
     listReposLinks (req, res, next) {
         console.log('checking for links on repo list');
 
-        let user = req.query.permission_user,
-            repos = req.entity;
-
-        if (user) {
-            repos.forEach((repo) => {
-                repo.links = [{
-                    rel: 'edit-user-permission',
-                    href: 'repos/' + repo.id + '/users/' + user + '/permissions/{permission}',
-                    method: 'PUT'
-                }, {
-                    rel: 'remove-user-permission',
-                    href: 'repos/' + repo.id + '/users/' + user,
-                    method: 'DELETE'
-                }];
+        let repos = req.entity,
+            username = req.query.permission_user,
+            permissions; //TODO: get permissions got logged in uer
+
+        if (username && permissions) {
+            repos.forEach(repo => {
+                let permission = permissions[repo.id];
+                if (permission.permissive === 'admin' || permission.github === 'admin') {
```

Comment:
The check for username indicates the list of repos is specific to a user, requiring links with valid permission.
The check for permissions prevents it from running until we get the logged in users permissions.